### PR TITLE
kms: use different context for server lifecycle and initial load

### PR DESCRIPTION
**Pull request by enj**
Signed-off-by: Monis Khan <mok@microsoft.com>

/kind bug

```release-note
Fixed a bug that resulted in "grpc: the client connection is closing" errors shortly after the Kubernetes API server automatically reloaded its encryption-at-rest config due to an observed change to the file.  This bug was only encountered when the --encryption-provider-config-automatic-reload flag was set to true.
```

### CODE REVIEW

File: test/integration/controlplane/transformation/kms_transformation_test.go [hunk 1]
```diff
@@ -706,6 +732,32 @@ resources:
...
+			_, err = test.restClient.CoreV1().Secrets("").List(
+				context.TODO(),
+				metav1.ListOptions{},
+			)
+			if err != nil {
+				t.Fatalf("failed to re-list secrets, err: %v", err)
+			}
+
+			// make sure things still work at a "later" time
+			time.Sleep(2 * time.Minute)
```
Comment (aojea) [inline test/integration/controlplane/transformation/kms_transformation_test.go:749]: this is going to add a big penalty 8 test-cases = 16 min, can't we do better?

Comment (enj) [reply test/integration/controlplane/transformation/kms_transformation_test.go:749]: I don't intend to merge this as-is, I am trying to root cause an issue I am seeing.

Comment (aojea) [reply test/integration/controlplane/transformation/kms_transformation_test.go:749]: oh, I was surprised , that explains it XD

File: test/integration/controlplane/transformation/kms_transformation_test.go [hunk 2]
```diff
@@ -706,6 +710,20 @@ resources:
 			if !bytes.HasPrefix(rawEnvelope, []byte(wantPrefix)) {
 				t.Fatalf("expected secret to be prefixed with %s, but got %s", wantPrefix, rawEnvelope)
 			}
+
+			// make sure things still work at a "later" time
+			time.Sleep(20 * time.Second)
```
Comment (enj) [inline test/integration/controlplane/transformation/kms_transformation_test.go:715]: This fails against master as expected:

```
    kms_transformation_test.go:718: Failed to create test secret, error: error while writing secret: Internal error occurred: rpc error: code = Canceled desc = grpc: the client connection is closing
I1116 21:02:22.232689 2038072 controller.go:211] Shutting down kubernetes service endpoint reconciler
E1116 21:02:22.233190 2038072 controller.go:214] Unable to remove endpoints from kubernetes service: StorageError: key not found, Code: 1, Key: /b36cdd15-dfb9-493c-b494-58d6e2f9667e/registry/masterleases//127.0.0.1, ResourceVersion: 0, AdditionalErrorMsg:
I1116 21:02:22.234391 2038072 cluster_authentication_trust_controller.go:463] Shutting down cluster_authentication_trust_controller controller
I1116 21:02:22.233642 2038072 controller.go:86] Shutting down OpenAPI V3 AggregationController
--- FAIL: TestEncryptionConfigHotReloadFileWatch (101.85s)
    --- FAIL: TestEncryptionConfigHotReloadFileWatch/truncate (25.35s)
    --- FAIL: TestEncryptionConfigHotReloadFileWatch/deleteAndCreate (25.58s)
    --- FAIL: TestEncryptionConfigHotReloadFileWatch/move (25.46s)
    --- FAIL: TestEncryptionConfigHotReloadFileWatch/symLink (25.45s)
FAIL
FAIL	k8s.io/kubernetes/test/integration/controlplane/transformation	102.041s
FAIL
```

Comment (liggitt) [reply test/integration/controlplane/transformation/kms_transformation_test.go:715]: can we limit this delay/check to one of the loop iterations (maybe the first one or the last one) so we don't expand the test runtime by 20seconds*testcases?

**Review by liggitt — COMMENTED** [review]
fix looks good, one request on the test

**Comment by liggitt** [comment]
/lgtm
/approve

since this will show up as a diff between rc0 and rc1, please add a release note indicating what was fixed

**Pull request merged**